docs/DafnyRef/Attributes.md

```diff
@@ -344,136 +343,215 @@ recursion was explicitly requested.
+Boogie usually produces an error message.  However, if at least one of
+the declarations is declared with :extern, one of the declarations is
+ignored.  If both declarations are :extern, Boogie arbitrarily chooses
+one of them to keep; otherwise, Boogie ignore the :extern declaration
+and keeps the other.
+
+#### 22.2.1.3. `{:checksum <string>}`
+
+Attach a checksum to be used for verification result caching.
+
+### 22.2.2. Verification attributes on functions and methods
+
+#### 22.2.2.1. `{:inline N}`
+     
+Inline given procedure (can be also used on implementation).
+N should be a non-negative number and represents the inlining depth.
+With `/inline:assume`, call is replaced with `assume false` once inlining depth is reached.
+With `/inline:assert`, call is replaced with `assert false` once inlining depth is reached.
+With `/inline:spec`, call is left as is once inlining depth is reached.
+With the above three options, methods with the attribute `{:inline N}` are not verified.
+With `/inline:none` the entire attribute is ignored.
+
+#### 22.2.2.2. `{:verify false}` {#sec-verify}
+     
+Skip verification of an implementation alltogether
+
+#### 22.2.2.3. `{:vcs_max_cost N}`
+#### 22.2.2.4. `{:vcs_max_splits N}`
+#### 22.2.2.5. `{:vcs_max_keep_going_splits N}`
+Per-implementation versions of
+`/vcsMaxCost`, `/vcsMaxSplits` and `/vcsMaxKeepGoingSplits`.
+
+#### 22.2.2.6. `{:selective_checking true}`
+Turn all asserts into assumes except for the ones reachable from
+assumptions marked with the attribute `{:start_checking_here}`.
+Thus, `assume {:start_checking_here} something;` becomes an inverse
+of `assume false;`: the first one disables all verification before
+it, and the second one disables all verification after.
+
+#### 22.2.2.7. `{:priority N}`
+Assign a positive priority 'N' to an implementation to control the order
+in which implementations are verified (default: N = 1).
+
+#### 22.2.2.8. `{:id <string>}`
+Assign a unique ID to an implementation to be used for verification
+result caching (default: "<impl. name>:0").
+
+#### 22.2.2.9. `{:timeLimit N}`
+Set the time limit for a given implementation.
+
+### 22.2.3. Verification attributes functions
+
+#### 22.2.3.1. `{:builtin "spec"}`
+#### 22.2.3.2. `{:bvbuiltin "spec"}`
+Rewrite the function to built-in prover function symbol 'fn'.
+
+#### 22.2.3.3. `{:inline}`
+`{:inline}` or `{:inline true}` expands function according to its definition before going to the prover.
+
+#### 22.2.3.4. `{:never_pattern true}`
+Terms starting with this function symbol will never be
+automatically selected as patterns. It does not prevent them
+from being used inside the triggers, and does not affect explicit
+trigger annotations. Internally it works by adding `{:nopats ...}`
+annotations to quantifiers.
+
+#### 22.2.3.5. `{:identity}`
+If the function annotated with `{:identity}` or `{:identity true}` has 1 argument and the use of it has type X->X for
```

Comment:
```suggestion
If the function annotated with `{:identity}` or `{:identity true}` has 1 argument and the use of it has type `X->X` for
```

Reply:
This was removed because it's Boogie-specific.